SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_ESPTraining.cpp

```diff
@@ -0,0 +1,211 @@
+/*  ESP Training
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "Common/Cpp/Exceptions.h"
+#include "Common/Cpp/PrettyPrint.h"
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "CommonFramework/Tools/StatsTracking.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "PokemonSV/Inference/PokemonSV_MainMenuDetector.h"
+#include "PokemonSV/Programs/PokemonSV_Navigation.h"
+#include "Pokemon/Inference/Pokemon_NameReader.h"
+#include "Pokemon/Pokemon_Strings.h"
+#include "PokemonSV/Inference/PokemonSV_OverworldDetector.h"
+#include "PokemonSV/Inference/Dialogs/PokemonSV_DialogDetector.h"
+#include "PokemonSV/Inference/PokemonSV_ESPEmotionDetector.h"
+#include "CommonFramework/VideoPipeline/VideoFeed.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/Tools/VideoResolutionCheck.h"
+#include "PokemonSV_ESPTraining.h"
+
+namespace PokemonAutomation {
+namespace NintendoSwitch {
+namespace PokemonSV {
+
+using namespace Pokemon;
+ 
+ESPTraining_Descriptor::ESPTraining_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonSV:ESPTraining",
+        STRING_POKEMON + " SV", "ESP Training",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonSV/ESPTraining.md",
+        "Clear the ESP Training to farm EV berries.",
+        FeedbackType::REQUIRED, true,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+struct ESPTraining_Descriptor::Stats : public StatsTracker {
+    Stats()
+        : m_emotions(m_stats["Emotions Shown"])
+        , m_joy(m_stats["Joy"])
+        , m_surprise(m_stats["Surprise"])
+        , m_excitement(m_stats["Excitement"])
+        , m_anger(m_stats["Anger"])
+        , m_clears(m_stats["Times Cleared"])
+        , errors(m_stats["Errors"])
+    {
+        m_display_order.emplace_back("Emotions Shown");
+        m_display_order.emplace_back("Joy");
+        m_display_order.emplace_back("Surprise");
+        m_display_order.emplace_back("Excitement");
+        m_display_order.emplace_back("Anger");
+        m_display_order.emplace_back("Times Cleared");
+        m_display_order.emplace_back(Stat("Errors"));
```

Comment:
Any reason to explicitly call the constructor here?

Reply:
Same as above, fixed in both this and chest farmer.